File: core/src/main/java/nl/t64/game/rpg/screens/world/pathfinding/TiledHeuristic.java
package nl.t64.game.rpg.screens.world.pathfinding;

import com.badlogic.gdx.ai.pfa.Heuristic;


class TiledHeuristic<N extends TiledNode> implements Heuristic<N> {

    @Override
    public float estimate(N node, N endNode) {
        return (float) Math.abs(endNode.x - node.x) + (float) Math.abs(endNode.y - node.y);
    }

}

File: core/src/nl/t64/game/rpg/screens/world/pathfinding/TiledHeuristic.kt
package nl.t64.game.rpg.screens.world.pathfinding

import com.badlogic.gdx.ai.pfa.Heuristic
import kotlin.math.abs


internal class TiledHeuristic : Heuristic<TiledNode> {

    override fun estimate(node: TiledNode, endNode: TiledNode): Float {
        return abs(endNode.x - node.x).toFloat() + abs(endNode.y - node.y).toFloat()
    }

}
